src/main/scala/za/co/absa/abris/avro/sql/AvroDataToCatalyst.scala

```diff
@@ -131,9 +147,7 @@ case class AvroDataToCatalyst(
 
   private def decodeVanillaAvro(payload: Array[Byte]): Any = {
 
-    decoder =  DecoderFactory.get().binaryDecoder(payload, 0, payload.length, decoder)
-    reader = new GenericDatumReader[Any](avroSchema)
-
-    reader.read(reader, decoder)
+    decoder = DecoderFactory.get().binaryDecoder(payload, 0, payload.length, decoder)
+    vanillaReader.read(result, decoder)
```

Comment:
Yes, the reader wasn't supposed to be there.

I would use null directly instead of using variable that is null anyway, 
or I would save the result of read to this variable, so it can be reused for the performance gains.

The reuse is described here
https://avro.apache.org/docs/1.8.2/gettingstartedjava.html#Deserializing

Reply:
Yes. I wanted to reuse the variable. Forgot to assign it back. Thanks! Fixed it.